### PR TITLE
erlang_ls 0.51.0 (new formula)

### PR DESCRIPTION
**Pull request by nikitaservais**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by nikitaservais** [comment]
Hello, I am sorry, but I do not understand the failed test `CI / macOS 14-arm64 (pull_request)`. It seems to be installing the formula without the dependency, but `escript` is needed for the installation, and it is in the dependency `erlang`. Could you give me feedback?

**Comment by SMillerDev** [comment]
You defined the dependency as build-only so it's uninstalling it for runtime testing. I guess you need to add a `depends_on "escript"`

**Review by chenrui333 — APPROVED** [review]

File: Formula/e/erlang_ls.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class ErlangLs < Formula
+  desc "Erlang Language Server"
+  homepage "https://erlang-ls.github.io/"
+  url "https://github.com/erlang-ls/erlang_ls/archive/refs/tags/0.51.0.tar.gz"
```
[the remark this answers is not in the document]
Comment (nikitaservais) [reply Formula/e/erlang_ls.rb:4]: Erlang uses snake_case, which may be the reason for its usage, can't know for sure.

[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/e/erlang_ls.rb:4]: yeah, I guess that might be the reason. (I see the artifact is also erlang_ls as well)

**Comment by chenrui333** [comment]
@nikitaservais, thanks for your first contribution to homebrew-core!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**